lib/logstash/filters/fingerprint.rb

```diff
@@ -62,7 +62,10 @@ def filter(event)
       when "PUNCTUATION"
         @source.sort.each do |field|
           next unless event.include?(field)
-          event[@target] = event[field].tr('A-Za-z0-9 \t','')
+          # In order to keep some backwards compatibility we should use the unicode version
+          # of the regexp because the POSIX one ([[:punct:]]) left some unwanted characters unfiltered (Symbols).
+          # gsub(/[^[:punct:]]/,'') should be equivalent to gsub(/[^[\p{P}\p{S}]]/,''), but not 100% in JRuby.
+          event[@target] = event[field].gsub(/[^[\p{P}\p{S}]]/,'')
```

Comment:
I did no recent test, but as far as I remember this is also does not work, as it translated to the other non posix version. Some edge case of this kind of expressions in ruby unicode. I'm curious, did you try it out like this?